2nd-gen/alerts-to-discord/functions/index.js

```diff
@@ -86,7 +91,49 @@ id: \`${id}\`
     // [END v2SendToDiscord]
   } catch (error) {
     logger.error(
-        `Unable to post fatal Crashlytics alert ${id} to Discord`,
+        `Unable to post fatal Crashlytics alert ${id} for ${appId} to Discord`,
+        error,
+    );
+  }
+});
+
+/**
+ * function triggered by AppDistribution that publishes a message
+ * to Discord whenever a new iOS tester device is registered.
+ */
+// [START v2AppDistributionAlertTrigger]
+exports.postnewduuidtodiscord = onNewTesterIosDevicePublished(async (event) => {
+// [END v2AppDistributionAlertTrigger]
+  // [START v2AppDistributionEventPayload]
+  // construct a helpful message to send to Discord
+  const appId = event.appId;
+  const {
+    testerDeviceIdentifier,
+    testerDeviceModelName,
+    testerEmail,
+    testerName,
+  } = event.data.payload;
+  const message = `
+📱 New iOS device registered by ${testerName} <${testerEmail}> for ${appId}
+
+UDID **${testerDeviceIdentifier}** for ${testerDeviceModelName}
+`;
```

Comment:
Developers will copy and paste this code - do we want to encourage them to share this much tester info into a team chat room? Would this sample still be useful/realistic if we only include `testerDeviceModelName` and `appId`, or maybe `testerName` as well?

Reply:
Good question! 

The message is really only actionable for developers who can update the provisioning profile and re-build and re-distribute an iOS app. So I'd expect this to be used to post to chat rooms with an appropriately limited audience. For other people the message would just be spam. For updating the provisioning profile, the developers need at least the App ID and the UDID (`${testerDeviceIdentifier}`) - the latter being the most sensitive value. So in order to be useful the message needs that data...

Or the example code gets modified to not post to Discord at all, but gets used as a basis to integrate with a CI/CD pipeline.

So I think it makes sense to extract *all* values from the payload just to demonstrate which values are available.